OsmAnd-java/src/main/java/net/osmand/router/HHRoutePlanner.java

```diff
@@ -174,6 +174,10 @@ public HHNetworkRouteRes runRouting(LatLon start, LatLon end, HHRoutingConfig co
 		if (hctx == null) {
 			return new HHNetworkRouteRes("Files for hh routing were not initialized. Route couldn't be calculated.");
 		}
+		hctx.rctx.mapIndexReaderFilter = new HashSet<>();
```

Comment:
1. move to initHCtx
2. This method requires to call RoutingContext.unloadAllData
3. That's why it needs to be checked whether there is something to filter or not. So it's better to create mapIndexExclude map so it's easy to check if there is something to exclude to clear cache.
4. **It's incorrect** doesn't include live update maps for same region